src/adapter/src/coord/sequencer/inner/copy_to.rs

```diff
@@ -6,59 +6,730 @@
+                }
+            };
+
+        // If we're in a multi-statement transaction and the query does not use `AS OF`,
+        // acquire read holds on any sources in the current time-domain if they have not
+        // already been acquired. If the query does use `AS OF`, it is not necessary to
+        // acquire read holds.
+        if in_immediate_multi_stmt_txn {
+            // Either set the valid read ids for this transaction (if it's the first statement in a
+            // transaction) otherwise verify the ids referenced in this query are in the timedomain.
+            if let Some(txn_reads) = self.txn_read_holds.get(session.conn_id()) {
+                // Find referenced ids not in the read hold. A reference could be caused by a
+                // user specifying an object in a different schema than the first query. An
+                // index could be caused by a CREATE INDEX after the transaction started.
+                let allowed_id_bundle = txn_reads.id_bundle();
+                let outside = source_bundle.difference(&allowed_id_bundle);
+                // Queries without a timestamp and timeline can belong to any existing timedomain.
+                if determination.timestamp_context.contains_timestamp() && !outside.is_empty() {
+                    let valid_names =
+                        self.resolve_collection_id_bundle_names(session, &allowed_id_bundle);
+                    let invalid_names = self.resolve_collection_id_bundle_names(session, &outside);
+                    return Err(AdapterError::RelationOutsideTimeDomain {
+                        relations: invalid_names,
+                        names: valid_names,
+                    });
+                }
+            } else {
+                if let Some((timestamp, bundle)) = potential_read_holds {
+                    let read_holds = self.acquire_read_holds(timestamp, bundle);
+                    self.txn_read_holds
+                        .entry(session.conn_id().clone())
+                        .or_insert_with(ReadHolds::new)
+                        .extend(read_holds);
+                }
+            }
+        }
+
+        // TODO (mouli): figure out what to do here
+
+        // let mut transaction_determination = determination.clone();
+        // if when.is_transactional() {
+        //     session.add_transaction_ops(TransactionOps::CopyTos {
+        //         determination: transaction_determination,
+        //         cluster_id,
+        //     })?;
+        // } else if matches!(session.transaction(), &TransactionStatus::InTransaction(_)) {
+        //     // If the query uses AS OF, then ignore the timestamp.
+        //     transaction_determination.timestamp_context = TimestampContext::NoTimestamp;
+        //     session.add_transaction_ops(TransactionOps::CopyTos {
+        //         determination: transaction_determination,
+        //         cluster_id,
+        //     })?;
+        // };
```

Comment:
This is copied from peeks. I am not exactly sure what this does or if this will be needed for COPY TO